### PR TITLE
Draft of Zfh extension for IEEE 754 binary16 support

### PR DESCRIPTION
**Pull request by aswaterman**
This proposal follows the template of the existing FP extensions.  Future extensions might also provide widening ops, but I think that's out of scope for the baseline.

### CODE REVIEW

**Comment by aswaterman** [comment]
See https://github.com/riscv/riscv-v-spec/issues/349

**Comment by kito-cheng** [comment]
Any plan for merge this? I think it's just 0.1 that mean we still have chance to change anything? otherwise it's like a hidden/secret extension in branch.

When I said there is already have a spec draft for zfh/fp16 and then people told me they don't seem the spec, that happened several times, and then I need to point out this is not exist in master branch yet, you need to find it out in `zfh` branch :P

**Comment by aswaterman** [comment]
I would love to merge it, but IIRC the plan was to wait until @kasanovic wrote an email proposing it.

**Comment by kito-cheng** [comment]
Oh, OK, thanks for clarification :)

**Comment by vowstar** [comment]
I really hope this PR could merge and freeze soon, this is a very useful extension.
I wonder if there is a corresponding extension for bfp16?

File: src/zfh.tex [hunk 1]
```diff
@@ -348,3 +348,46 @@ \section{Half-Precision Floating-Point Classify Instruction}
...
+FSGNJ.S is used to move half-precision values between floating-point registers.
+\end{commentary}
+
+\begin{commentary}
+Half-precision addition, subtraction, multiplication, division, and
+square-root operations can be faithfully emulated by converting the
+half-precision operands to single-precision, performing the operation
+using single-precision arithmetic, then converting back to half-precision.
+Performing half-precision fused multiply-addition using this method incurs
+a 1-ulp error on some inputs.
```
Comment (knightsifive) [inline src/zfh.tex:384]: Suggest clarifying that the "error" is only on RNE and RMM (double-rounding is innocuous for directed rounding).

Comment (aswaterman) [reply src/zfh.tex:384]: @knightsifive Yeah, good point.

**Pull request merged**